src/LoginDialog/OidcHandler.tsx

```diff
@@ -6,40 +6,41 @@ import { generateRedirectUri } from './OidcUtilities'
 
 export const OidcHandler: React.FC<any> = ()=> {
 
-    const history = useHistory()
-    const location = useLocation()
-    const { setCustomerData } = useCustomerData();
-
-    useEffect(() => {
-        async function setCustomerDataFromOidcCallback() {
-            const redirectInitialLocation:string = localStorage.getItem('location') || '/';
-
-            let query = new URLSearchParams(location.search);
-            const code = query.get('code')
-            const state = query.get('state')
-            const codeVerifier = localStorage.getItem('code_verifier');
-
-            if(code !== undefined && state !== undefined) {
-                if (state === localStorage.getItem('state') && typeof codeVerifier === "string" ) {
-
-                    const response: any = await oidcLogin(code!, generateRedirectUri(), codeVerifier)
-                    const result = response;
-                    
-                    setCustomerData(result.token, result.customer_id);
-                    
-                    history.push(redirectInitialLocation);
-                } else {
-                    alert('Unable to validate identity');
-                    history.push(redirectInitialLocation);
-                }
-
-                localStorage.removeItem('location')
-                localStorage.removeItem('state')
-            }
+  const history = useHistory();
+  const historyPush = history.push;
+  const location = useLocation();
+  const locationSearch = location.search;
+  const { setCustomerData } = useCustomerData();
+
+  useEffect(() => {
+    async function setCustomerDataFromOidcCallback() {
+      const redirectInitialLocation: string = localStorage.getItem('location') || '/';
+
+      let query = new URLSearchParams(locationSearch);
+      const code = query.get('code')
+      const state = query.get('state')
+      const codeVerifier = localStorage.getItem('code_verifier');
+
+      if (code !== undefined && state !== undefined) {
+        if (state === localStorage.getItem('state') && typeof codeVerifier === "string" ) {
+
+          const response: any = await oidcLogin(code!, generateRedirectUri(), codeVerifier)
+          const result = response;
+          setCustomerData(result.token, result.customer_id);
+
+          historyPush(redirectInitialLocation);
+        } else {
+          alert('Unable to validate identity');
```

Comment:
I just noticed. What's the purpose of this alert? It's also in english and not translated. Should this be in the console or is it intended to show in the user's flow as per UX?

Reply:
This only happens if someone was tampering with the codes returned from oidc provider and manually setting them in url. It should never happen during the normal use.